src/components/molecules/Visualizer/Engine/Cesium/Box/index.tsx

```diff
@@ -0,0 +1,172 @@
+    planeDimensions.x = boxDimensions.x;
+    planeDimensions.y = boxDimensions.z;
+  } else if (planeNormalAxis === Axis.Z) {
+    planeDimensions.x = boxDimensions.x;
+    planeDimensions.y = boxDimensions.y;
+  }
+}
+
+const Side: FC<{
+  id: string;
+  planeLocal: CesiumPlane;
+  trs: TranslationRotationScale;
+  style: BoxStyle<Color>;
+}> = memo(
+  function SidePresenter({ id, planeLocal, style, trs }) {
+    const normalAxis = planeLocal.normal.x ? Axis.X : planeLocal.normal.y ? Axis.Y : Axis.Z;
+    const cbRef = useMemo(
+      () => new CallbackProperty(() => trs.translation, false) as unknown as PositionProperty,
+      [trs],
+    );
+    const [plane, dimension] = useMemo(() => {
+      const dimension = new Cartesian3();
+      setPlaneDimensions(trs.scale, normalAxis, dimension);
+      const scratchScaleMatrix = new Matrix4();
+      const scaleMatrix = Matrix4.fromScale(trs.scale, scratchScaleMatrix);
+      const plane = CesiumPlane.transform(
+        planeLocal,
+        scaleMatrix,
+        new CesiumPlane(Cartesian3.UNIT_Z, 0),
+      );
+      return [plane, dimension];
+    }, [trs.scale]);
+
+    return (
+      <Entity id={id} position={cbRef}>
+        <PlaneGraphics
+          plane={plane}
+          dimensions={dimension}
+          fill={style.fill}
+          outline={style.outline}
+          material={style.fillColor}
+          outlineColor={style.outlineColor}
+          outlineWidth={style.outlineWidth}
+        />
+      </Entity>
+    );
+  },
+  () => true,
+);
+
+// The 6 box sides defined as planes in local coordinate space.
+// ref: https://github.com/TerriaJS/terriajs/blob/cad62a45cbee98c7561625458bec3a48510f6cbc/lib/Models/BoxDrawing.ts#L161-L169
+const SIDE_PLANES: readonly CesiumPlane[] = [
+  new CesiumPlane(new Cartesian3(0, 0, 1), 0.5),
+  new CesiumPlane(new Cartesian3(0, 0, -1), 0.5),
+  new CesiumPlane(new Cartesian3(0, 1, 0), 0.5),
+  new CesiumPlane(new Cartesian3(0, -1, 0), 0.5),
+  new CesiumPlane(new Cartesian3(1, 0, 0), 0.5),
+  new CesiumPlane(new Cartesian3(-1, 0, 0), 0.5),
+];
+
+// This is used for plane ID.
+// We can use this name, for example you want to move the box when front plane is dragged.
+const SIDE_PLANE_NAMES = ["bottom", "top", "front", "back", "left", "right"];
+
+const updateTrs = (trs: TranslationRotationScale, property: Property | undefined) => {
+  const { position, location, height, dimensions } = property?.default ?? {};
+
+  const pos = position || location;
+  const translation = pos ? Cartesian3.fromDegrees(pos.lng, pos.lat, height ?? 0) : undefined;
+  if (translation) {
+    Cartesian3.clone(translation, trs.translation);
+  }
+
+  // Quaternion.clone(trs.rotation, this.trs.rotation);
+
+  Cartesian3.clone(
+    new Cartesian3(dimensions?.x || 100, dimensions?.y || 100, dimensions?.z || 100),
+    trs.scale,
+  );
+
+  return trs;
+};
+
+const Box: React.FC<PrimitiveProps<Property>> = memo(function BoxPresenter({ layer }) {
```

Comment:
<PrimitiveProps<Property>> can just be `Props`, no? (above its declared like that)

Reply:
I think we should use PrimitiveProps.
PrimitiveProps indicates an interface which a reearth layer can take.

Reply:
Fair enough. If that is the case, is the exported Props type (on line 10) used in another component? If not, please remove the export. I see you import `Props` from `../../Primitive` in this component and do the same thing in `Tileset/index` so I am guessing it might not be necessary

Reply:
Yes, it is. The props is used in storybook for the box.